R/fa.R

```diff
@@ -225,9 +225,11 @@ fa <- function(name,
 
   svg <- HTML(svg)
 
-  class(svg) <- c("fontawesome", "svg", class(svg))
-
-  svg
+  structure(svg,
```

Comment:
We are trying to get away from using `structure()` in our code. Winston found that it's actually slower than using `list()` and `<-class`.

Reply:
Yes, `structure()` is surprisingly slow, so it's generally better to use `class()<-`.

I often use a utility function called `set_class()`: 

```R
set_class <- function(x, class) {
  class(x) <- class
  x
}

x <- list()

microbenchmark::microbenchmark(
  structure(x, class = "foo"),
  set_class(x, "foo")
)
#> Unit: nanoseconds
#>                         expr  min     lq     mean median     uq     max neval
#>  structure(x, class = "foo") 3388 3492.5  3910.12 3550.5 3653.5   20464   100
#>          set_class(x, "foo")  821  869.5 16954.03  904.5  935.5 1598836   100
```

Reply:
Oh, but since you're also adding other attributes, it makes sense to use `structure()` here.